gui/src/components/preboarding/login.vue

```diff
@@ -86,110 +86,15 @@
-                :label="$t('common.new-password-label')"
-                :message="passwordTooltipMessage"
-              />
-            </label>
-            <input
-              class="cortx-form__input_text"
-              type="password"
-              id="user-password"
-              name="user-password"
-              v-model.trim="resetAccountForm.password"
-              @input="$v.resetAccountForm.password.$touch"
-            />
-            <div class="cortx-form-group-label cortx-form-group-error-msg">
-              <label
-                id="new-password-required-error"
-                v-if="
-                  $v.resetAccountForm.password.$dirty &&
-                  !$v.resetAccountForm.password.required
-                "
-                >{{ $t("common.password-required") }}</label
-              >
-              <label
-                id="new-password-invalid-error"
-                v-else-if="
-                  $v.resetAccountForm.password.$dirty &&
-                  !$v.resetAccountForm.password.passwordRegex
-                "
-                >{{ $t("common.invalid-password") }}</label
-              >
-            </div>
-          </div>
-        </v-col>
-        <v-col class="col-6 ml-7 pt-0">
-          <div
-            class="cortx-form-group"
-            :class="{
-              'cortx-form-group--error':
-                $v.resetAccountForm.confirmPassword.$error
-            }"
-          >
-            <label
-              class="cortx-form-group-label"
-              for="confirm-password"
-              id="confirm-password-label"
-              >{{ $t("common.confirm-password-label") }}</label
-            >
-            <input
-              class="cortx-form__input_text"
-              type="password"
-              id="confirm-password"
-              name="confirm-password"
-              v-model.trim="resetAccountForm.confirmPassword"
-              @input="$v.resetAccountForm.confirmPassword.$touch"
-            />
-            <span
-              id="confirm-password-notmatch-error"
-              class="cortx-form-group-label cortx-form-group-error-msg"
-              v-if="
-                $v.resetAccountForm.confirmPassword.$dirty &&
-                !$v.resetAccountForm.confirmPassword.sameAsPassword
-              "
-              >{{ $t("common.password-not-match") }}</span
-            >
-          </div>
-        </v-col>
-        <v-col class="col-6 ml-7 pb-6 pt-0">
-          <button
-            type="button"
-            id="reset-password-button"
-            class="cortx-btn-primary"
-            @click="resetPassword()"
-            :disabled="$v.resetAccountForm.$invalid"
-          >
-            {{ $t("login.reset-btn") }}
-          </button>
-        </v-col>
-      </v-card>
-    </v-dialog>
-      
+
+    <cortx-reset-password
+      v-if="showResetPasswordDialog"
+      :showResetPasswordDialog="showResetPasswordDialog"
+      :authToken="authToken"
```

Comment:
Auth token is an application level state should not be passed around in components.
Please create a isUserAuthorized property in user-login store and create a getter and setter(mutation) for the same.
1. After you make login api call and get authToken, set it in the localstorage.
2. Show the reset password popup. Reset password api will be called (authToken will be available for the api).
3. After password reset API set isUserAuthorized to true.
4. In router.ts line no 416 Replace if(!token) with if(!store.getIsAuthorized)
5. In logout.ts mixin store.setIsAuthorized(false)

This is a high level workflow and psuedo code. Please make changes as required and test all scenarios.

Reply:
Updated logic which removes passing auth token and username. Added setters and getters to set user authorized property.